### PR TITLE
(POC) atomic: Refactored facet value and base facet to use functional stencil components

### PR DESCRIPTION
**Pull request by nrawji**
An example of how the "controller" system that @ThibodeauJF suggested in stand-up would work, let me know what you think.

### CODE REVIEW

File: packages/atomic/src/components/facets/atomic-category-facet/atomic-category-facet.pcss [hunk 1]
```diff
@@ -1,6 +1,3 @@
 @import "tailwindcss/base";
 @import "tailwindcss/utilities";
-
-:host {
-  display: block;
-}
+@import "../facet/base-facet.pcss";
```
Comment (nrawji) [inline packages/atomic/src/components/facets/atomic-category-facet/atomic-category-facet.pcss:3]: There's no way to import a css file into a functional component, so each base component would need to import the styles themselves.

File: packages/atomic/src/components/facets/facet/base-facet.tsx [hunk 2]
```diff
@@ -1,82 +1,81 @@
...
-  tag: 'base-facet',
-  styleUrl: 'base-facet.pcss',
-  shadow: false,
-})
-export class BaseFacet {
-  @State() public isExpanded: Boolean = false;
-  @Event() public deselectAll!: EventEmitter<void>;
-  @Prop() public label!: string;
-  @Prop() public hasActiveValues!: boolean;
+export interface BaseFacetState {
```
Comment (nrawji) [inline packages/atomic/src/components/facets/facet/base-facet.tsx:5]: Each base component implements BaseFacetState and then passes itself into the controller, the controller is then passed into the functional component which can now mutate state on its parent.

File: packages/atomic/src/components/facets/atomic-date-facet/atomic-date-facet.tsx [hunk 3]
```diff
@@ -52,11 +59,11 @@ export class AtomicDateFacet implements InitializableComponent {
     const isSelected = this.facet.isValueSelected(item);
 
     return (
-      <facet-value
+      <FacetValue
```
Comment (samisayegh) [inline packages/atomic/src/components/facets/atomic-date-facet/atomic-date-facet.tsx:62]: Would it be possible to keep the same casing by decorating the function?

**Review by samisayegh — APPROVED** [review]
More of a general curiosity, were there any benefits of using functional stencil components? And how do they compare to using state components (or class instances) everywhere for consistency?

**Review by btaillon — APPROVED** [review]
Looks good! If you haven't yet, I would recommend testing this on Firefox since SVG icons don't seem to always behave the same on Firefox.

**Comment by ThibodeauJF** [comment]
I like this approach! We could go even farther and make facet-search a functionnal component by leveraging `CustomEvent` like in `initialization-utils.tsx` line 80

**Review by ThibodeauJF — APPROVED** [review]

**Comment by nrawji** [comment]
> More of a general curiosity, were there any benefits of using functional stencil components? And how do they compare to using state components (or class instances) everywhere for consistency?

The biggest benefit is they're truly internal, they don't generate their own web components and are inaccessible externally.

**Pull request closed without merging**